### DISCUSSION TITLE
HomeKit tile to schedule dishwasher

### DISCUSSION THREAD

**Discussion by MathieuN** [thoukydides/homebridge-homeconnect#168] [Ideas]
Have a Bosch dishwasher connected to home connect and home bridge. Would like to create a HomeKit tile to schedule a wash at certain time of the day (to take advantage of cheap electric price at night) with a certain preset. Is it possible?

**Comment by thoukydides** [comment]
Yes, if your appliance supports either the `BSH.Common.Option.StartInRelative` or `BSH.Common.Option.FinishInRelative` option for the programme that you want to start. Rather than specifying a delay in seconds for that value use a time in `HH:MM` format. This capability has been supported since v0.23.0.

**Reply by MathieuN** [reply]
Thanks good news. How would I do that then and how to check if the device supports those options?

cheers

**Comment by thoukydides** [comment]
It should all be explained in the documentation:
* [config.json](https://github.com/thoukydides/homebridge-homeconnect/wiki/config.json)
* [Customising Appliance Programs](https://github.com/thoukydides/homebridge-homeconnect/wiki/Programs)

If you are using [homebridge-config-ui-x](https://github.com/oznu/homebridge-config-ui-x) then:
1. Open the plugin's settings.
2. Expand the `Dishwasher` options.
3. Ensure that Program Switches is set to `Custom list of programs and options`.
4. Configure the program options that you require, entering the required time in the `Start time` or `Finish time` field.
5. Save the modified settings.

Otherwise invoke the HomeKit `Identify` routine for the appliance and look in the Homebridge log file to see the supported options, and then manually configure the required settings in the `config.json` file.

According to the Home Connect API documentation:
* [`BSH.Common.Option.StartInRelative`](https://api-docs.home-connect.com/programs-and-options/?#dishwasher_start-in-relative-option) can be supported by Dishwasher and Oven appliances.
* [`BSH.Common.Option.FinishInRelative`](https://api-docs.home-connect.com/programs-and-options/?#washer_finish-in-relative-option) can be supported by Washer, Dryer, and WasherDryer, appliances.

However, not all appliances of these types will necessarily support these options. Even appliances that do support them may not allow their use for all programme types.

In my case, my [Siemens iQ700 Dishwasher](https://www.siemens-home.bsh-group.com/uk/mysiemens/products/0004436388) (SN678D06TG/53) does not support delayed start at all. However, my [Siemens iQ700 Single Oven](https://www.siemens-home.bsh-group.com/uk/mysiemens/products/0004401572) (HB678GBS6B/58) does, but only for some of its programmes. Here is an example of how I use this capability:

![image](https://github.com/thoukydides/homebridge-homeconnect/assets/4775242/7df8ad09-97de-465d-a4f8-f48589bb949c)

```
            "SIEMENS-HB678GBS6B-123456789ABC": {
                "addprograms": "custom",
                "programs": [
                    {
                        "name": "Baked Potato",
                        "key": "Cooking.Oven.Program.HeatingMode.HotAir",
                        "selectonly": false,
                        "options": {
                            "Cooking.Oven.Option.SetpointTemperature": 180,
                            "BSH.Common.Option.Duration": 5400,
                            "Cooking.Oven.Option.FastPreHeat": false,
                            "BSH.Common.Option.StartInRelative": "09:30"
                        }
                    }
                ]
            },
```
